internal/config/config.go

```diff
@@ -90,43 +101,44 @@ func (c *cfg) Authentication() gh.AuthConfig {
 	return &AuthConfig{cfg: c.cfg}
 }
 
-func (c *cfg) Browser(hostname string) string {
-	val, _ := c.GetOrDefault(hostname, browserKey)
-	return val
+func (c *cfg) Browser(hostname string) gh.ConfigEntry {
+	// Intentionally panic as this is a programmer error
```

Comment:
I think this is much nicer than ignoring the error and having an unexpected empty string.

Reply:
How would you feel about rephrasing this so the reader realizes this might panic under certain conditions rather than will panic in all conditions?

```suggestion
	// Intentionally choosing for this to panic if browser has not been configured
```

Reply:
As an aside, what would the user or programmer do if/when `gh` panics because of this or the other situations that follow?

Reply:
Well, if this panics it's because someone has referenced a key that is supposed to [have a default value](https://github.com/cli/cli/blob/7eddd537ba74e90ea141fe2f4daadbf6b9d01273/internal/config/config.go#L525) but doesn't. As a programmer we expect these to have default values, so it's better to blow up than do something surprising. The programmer should make sure the key exists. [Note that we have tests covering the existing ones, though they could be made a little more black box.](https://github.com/cli/cli/blob/7eddd537ba74e90ea141fe2f4daadbf6b9d01273/internal/config/config_test.go#L131)